### PR TITLE
Create v4.0.2 and pin galsim and stomp.py

### PR DESCRIPTION
**Pull request by mwittgen**
Checklist
* [X] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [X] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by ktlim** [comment]
@conda-forge-admin, please rerender

**Comment by ktlim** [comment]
You don't want to add a `stomp.py >= 8.0.1` pin?

File: recipe/meta.yaml [hunk 1]
```diff
@@ -144,7 +144,8 @@ outputs:
         - iminuit >=2
         - lmfit !=1.0.3
         - lsstdesc.coord
-        - matplotlib-base
+        # matplotlib-base 3.6 breaks fgcm; see DM-36267
+        - matplotlib-base <3.6
```
Comment (erykoff) [inline recipe/meta.yaml:148]: This is fine, but I believe that the bug in matplotlib was fixed for 3.6.1.  Have you seen this as a problem now?  Or could we do `!=3.6.0`?

**Review by erykoff — APPROVED** [review]
I think it's unfortunate that we called the branch `4.0.1` and not `4.0.x`...

File: recipe/meta.yaml (hunk 1 above)
Comment (mwittgen) [reply recipe/meta.yaml:148]: I tested successfully with {{3.6.1}}

Comment (mwittgen) [reply recipe/meta.yaml]: Unpinned `matplotlib-base`

File: .ci_support/osx_64_.yaml [hunk 2]
```diff
@@ -31,7 +31,7 @@ fftw:
 fortran_compiler:
 - gfortran
 fortran_compiler_version:
-- '9'
+- '11'
```
Comment (erykoff) [inline .ci_support/osx_64_.yaml:34]: This change from the rerenderer is breaking osx-64.  I don't understand it, but we should figure this out for 5.0.0.

**Comment by erykoff** [comment]
Is it a problem that we don't specifically exclude matplotlib-base=3.6.0?

**Comment by mwittgen** [comment]
> Is it a problem that we don't specifically exclude matplotlib-base=3.6.0?

Do you mean `matplotlib-base!=3.6.0` ? `3.6.1` and `<3.6` work

**Comment by erykoff** [comment]
My phrasing was poor, yes you are correct.

**Comment by mwittgen** [comment]
Is there anything missing to have 4.0.2 be made available?

**Comment by erykoff** [comment]
I don't know of anything else, and if it solves and builds right now then I think it's sufficient, unless we put upper limits on everything which hasn't been our practice.  @ktlim, your thoughts?

**Pull request merged**